plugin/arcconf/arcconf.py

```diff
@@ -0,0 +1,698 @@
+                        ld_info = \
+                            decoded_json['Controller']['LogicalDrive'][ld]
+                        lsm_vol = \
+                            Arcconf._arcconf_ld_to_lsm_vol(ld_info,
+                                                           pool_id,
+                                                           sys_id,
+                                                           cnt,
+                                                           str(ld_num),
+                                                           ld_name)
+                        lsm_vols.append(lsm_vol)
+            except DeviceNotFoundError:
+                pass
+
+        return search_property(lsm_vols, search_key, search_value)
+
+    @staticmethod
+    def _arcconf_disk_to_lsm_disk(arcconf_disk, sys_id, ctrl_num):
+        disk_id = arcconf_disk['serialNumber'].strip()
+
+        disk_name = "%s" % (arcconf_disk['model'])
+        disk_type = _disk_type_of(arcconf_disk)
+        link_type = disk_type
+
+        try:
+            blk_size = int(arcconf_disk['physicalBlockSize'])
+        except KeyError:
+            blk_size = int(arcconf_disk['blockSize'])
+        blk_count = int(arcconf_disk['numOfUsableBlocks'])
+
+        status = _disk_status_of(arcconf_disk)
+        disk_reported_slot = arcconf_disk['fsaSlotNum']
+        disk_channel = arcconf_disk['channelID']
+        disk_device = arcconf_disk['deviceID']
+        disk_location = \
+            "%s %s %s" % (disk_reported_slot, disk_channel, disk_device)
+        plugin_data = \
+            "%s,%s,%s,%s" % (ctrl_num, disk_channel, disk_device, status)
+
+        # TODO(Raghavendra) Need to find better way of getting the vpd83 info.
+        # Just providing disk_path did not yield the correct vpd info.
+        vpd83 = ''
+        rpm = arcconf_disk['rotationalSpeed']
+
+        return Disk(
+            disk_id, disk_name, disk_type, blk_size, blk_count,
+            status, sys_id, _plugin_data=plugin_data, _vpd83=vpd83,
+            _location=disk_location, _rpm=rpm, _link_type=link_type)
+
+    @_handle_errors
+    def disks(self, search_key=None, search_value=None,
+              flags=Client.FLAG_RSVD):
+
+        rc_lsm_disks = []
+        controllerList = []
+
+        getconfig_cntrls_info = self._get_detail_info_list()
+        sys_id = ''
+        cntrl = 0
+
+        for decoded_json in getconfig_cntrls_info:
+            sys_id = cntrl + 1
+            cntrl_num = int(cntrl + 1)
+            cntrl += 1
+
+            for channel in range(len(decoded_json['Controller']['Channel'])):
+                channel_range = decoded_json['Controller']['Channel'][channel]
+                if 'HardDrive' in channel_range:
+                    hd_disks_num = \
+                        len(decoded_json['Controller']
+                                        ['Channel']
+                                        [channel]
+                                        ['HardDrive'])
```

Comment:
Please use `channel_range` instead of `decoded_json['Controller']['Channel'][channel]` here.

There are also similar lines below.